app/presenters/content_item/contents_list.rb

```diff
@@ -46,15 +46,33 @@ def first_item_has_long_content?
 
     def first_item_content
       element = first_item
-      first_item_text = ""
-      allowed_elements = %w[p ul ol]
+      return "" unless element
 
-      until element.name == "h2"
-        first_item_text += element.text if element.name.in?(allowed_elements)
+      content = ""
+
+      until element.nil? || element.name == "h2"
+        content += extract_content_from_element(element)
         element = element.next_element
-        break if element.nil?
       end
-      first_item_text
+
+      content.strip
+    end
+
+    def extract_content_from_element(element)
+      case element.name
+      when "div"
+        extract_nested_content(element)
+      when "p", "ul", "ol"
+        element.text
+      else
+        ""
+      end
+    end
+
+    def extract_nested_content(element)
```

Comment:
This is some nice fancy code!! I think we should check that this is behaviour we actually want though first. I'm wondering if it might result in longer than expected contents lists as we nibble through every element on the page. Do you happen to have an example of a page with very nested content so we can look at the before an after?